### DISCUSSION TITLE
How can we implement the xTTSv2 model on the OpenWebUI interface?

### DISCUSSION THREAD

**Discussion by xyz-rainbow** [coqui-ai/TTS#3795] [General Q&A]
we need a URL and API key
![image](https://github.com/coqui-ai/TTS/assets/164746451/a3ba9816-8c40-4636-90a0-f0e6f09be9ef)

the openwebui proyect; https://github.com/open-webui/open-webui

+ Im on windows, using the "docker desktop"
+ The Openwebui works on the docker
+ the tts should work on the docker too like the "openedai-speech-integration" https://docs.openwebui.com/tutorial/openedai-speech-integration/

+LOCAL OPENWEBUI URL: http://localhost:3000/
+LOCAL OPENEDAAITTS: http://localhost:8000/

**Comment by hgruber** [comment]
Until there's a better solution I helped myself with a quick hack:
As I have haproxy already running I did a quick open.ai to coqui api rewrite:

```
frontend tts
    bind *:8001
    mode http
    use_backend coqui

backend coqui
    mode http
    option http-buffer-request
    http-request set-path /api/tts?text=%[req.body,json_query('$.input'),regsub('\"','','g'),url_enc]
    http-request set-method GET
    server local_coqui 127.0.0.1:5002
```

so in OpenWebUI point your TTS to `http://yourserver:8001`
With OpenWebUI's more recent release I have character encoding issues. OpenWebUI sends utf8 while coqui seems to expect ISO-8859-1. Apart from that it works.
